src/MovingImage/Client/VMPro/Manager/TokenManager.php

```diff
@@ -108,14 +107,39 @@ protected function createNewTokens()
     /**
      * Create a new access token for a video manager using a refresh token.
      *
-     * @TODO Implement this..
-     *
      * @param Token $refreshToken
      * @param int   $videoManagerId
+     *
+     * @return Token
      */
     protected function createAccessTokenFromRefreshToken(Token $refreshToken, $videoManagerId)
     {
-        throw new Exception('Using refresh tokens is currently not implemented..');
+        $logger = $this->getLogger();
+        $logger->debug('Starting request to create fresh access token from refresh token');
+
+        try {
+            $response = $this->httpClient->post(sprintf('auth/refresh/%d', $videoManagerId), [
+                'json' => [
+                    'refreshToken' => $refreshToken->getTokenString(),
+                ],
+                'headers' => [
+                    'accept: application/json',
+                    'cache-control: no-cache',
+                    'content-type: application/json',
+                ],
+            ]);
+
+            $data = \json_decode($response->getBody(), true);
+            $logger->debug('Successfully retrieved new access token', $data);
+
+            return new Token(
+                $data['accessToken'],
+                $this->tokenExtractor->extract($data['accessToken']),
+                $videoManagerId
+            );
+        } catch (\Exception $e) {
+            throw $e;
```

Comment:
I'm removing this no-op error handling until we implement scoped exceptions at a later time - pushing a change in a minute